### PR TITLE
compat tidyup

### PR DESCRIPTION
**Pull request by ldir-EDB0**
This needs a sanity check but here's a commit that IMO tidies up some of the compatibility cruft.  It's been successfully compiled on 3.16 & 4.9, hence solves https://github.com/dtaht/sch_cake/issues/91

### CODE REVIEW

**Comment by Lochnair** [comment]
Still not happy on the 3.10.107 kernel.
```
/var/lib/jenkins/workspace/ubiquiti/bufferbloat/cake_cleanup/sch_cake.c: In function 'cake_drop':
/var/lib/jenkins/workspace/ubiquiti/bufferbloat/cake_cleanup/sch_cake.c:1313:2: error: implicit declaration of function 'qdisc_tree_reduce_backlog' [-Werror=implicit-function-declaration]
  qdisc_tree_reduce_backlog(sch, 1, len);
  ^

```

Rest of the build log [here](https://build.lochnair.net/job/ubiquiti/job/bufferbloat/job/cake_cleanup/6/console).

**Comment by ldir-EDB0** [comment]
I'm open to suggestions.  I'd like to think it'd be easier now all the compatibility cruft is in cobalt_compat.h. - I've been banging my head against this for a day and a half so far... need a rest :-)

**Comment by ldir-EDB0** [comment]
have another try

**Comment by Lochnair** [comment]
Alright, compiles now, thank you :)
I can try it on my ER-X just to be sure, after I've got the tc-adv builds running again.

**Comment by ldir-EDB0** [comment]
@Lochnair Hi Nils, I've made this change slightly less invasive to sch_cake.c - can you check that it still compiles for you.

File: cobalt_compat.h [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+static inline void qdisc_qstats_drop(struct Qdisc *sch)
+{
+	sch->qstats.drops++;
+}
+
+#endif
+
+#if KERNEL_VERSION(4, 1, 0) > LINUX_VERSION_CODE
+#if !defined(IS_REACHABLE)
```
Comment (tohojo) [inline cobalt_compat.h]: You can remove the !defined check if you are checking the kernel version anyway... Or leave it and drop the kernel version check, I guess...

Comment (ldir-EDB0) [reply cobalt_compat.h]: yeah I bottled out of removing the double check 'cos I hadn't gone through and checked the kernel version didn't have some sub version exceptions.

I'll work up the enthusiasm to do a thorough check :-)

Comment (tohojo) [reply cobalt_compat.h]: well it's a macro, so there shouldn't be any problems in just keeping the !defined check and dropping the kernel version check...

**Pull request merged**